src/python/antares_xpansion/input_checker.py

```diff
@@ -260,27 +316,106 @@ def check_candidates_file(driver):
             elif value == '1':
                 has_a_profile = True
             else:
+                # check file existence
+                filename_path = os.path.normpath(
+                    os.path.join(capacity_dir_path, value))
+                if not os.path.isfile(filename_path):
+                    flushed_print('Illegal value : option can be 0, 1 or an existent filename.\
+                            %s is not an existent file' % filename_path)
+                    raise ProfileFileNotExists
                 has_a_profile = has_a_profile or check_profile_file(
-                    driver.capacity_file(value))
+                    filename_path)
         if not has_a_profile:
             # remove candidate if it has no profile
             flushed_print("candidate %s will be removed!" %
                           ini_file[each_section]["name"])
             ini_file.remove_section(each_section)
             config_changed = True
 
-    if config_changed:
-        shutil.copyfile(driver.candidates_ini_filepath(),
-                        driver.candidates_ini_filepath() + ".bak")
-        with open(driver.candidates_ini_filepath(), 'w') as out_file:
-            ini_file.write(out_file)
-        flushed_print("%s file was overwritten! backup file %s created"
-                      % (driver.candidates_ini_filepath(), driver.candidates_ini_filepath() + ".bak"))
+    return config_changed
+
+
+def check_attributes_profile(ini_file, candidates_ini_filepath, capacity_dir_path):
+    # check attributes profile is 0, 1 or an existent filename
+    if check_attribute_profile_values(ini_file, capacity_dir_path):
+        copy_in_backup(ini_file, candidates_ini_filepath)
+
+
+def check_candidates_file(candidates_ini_filepath, capacity_dir_path):
+    """
+        checks that a candidate file related to an XpansionDriver has the correct format
+
+        :param driver: the XpansionDriver pointing to the candidates file
+
+        :return: Exits if the candidates files has the wrong format.
+    """
+    default_values = {'name': 'NA',
```

Comment:
I find it weird that the deprecated options 
   - `enable`
   - `candidate-type`
   - `investment-type`
   - `relaxed`
   - `has-link-profile`
  
 appear here. Are they necessary for the `ConfigParser` initialization and can they be removed ?

Reply:
Yes they are checked to inform user to not use them.